serverless/aws/functions/generic.py

```diff
@@ -101,14 +98,19 @@ def __init__(
         if use_async_dlq:
             self.use_async_dlq()
 
-        log_group = dict(Type="AWS::Logs::LogGroup", Properties=dict(RetentionInDays=30))
+        log_group_properties = kwargs.pop("log_group", {}).get("Properties") if kwargs.get("log_group", {}).get("Properties") else dict(RetentionInDays=30)
+
+        log_group = dict(Type="AWS::Logs::LogGroup", Properties=log_group_properties)
```

Comment:
I think there might be a bug here if you have a `log_group` argument containing an "falsey" `Properties` value. It wouldn't pop the `log_group` argument in that case, and then it would get included in the loop at the end of the function.

I think this would achieve basically the same thing:

```python
log_group_properties = kwargs.pop("log_group", {}).get("Properties") or dict(RetentionInDays=30)
```